mux_test.go

```diff
@@ -1967,6 +1967,314 @@ func TestErrMatchNotFound(t *testing.T) {
 	}
 }
 
+// methodsSubrouterTest models the data necessary for testing handler
+// matching for subrouters created with set HTTP methods.
+type methodsSubrouterTest struct {
+	title      string
+	router     *Router
+	flags      []bool
+	methods    []string
+	method     string
+	path       string
+	redirectTo string
+	wantCode   int
+}
+
+// TestMethodsSubrouter tests the correct matching of handlers for
+// subrouters registered after a route's method-matcher is set.
+func TestMethodsSubrouter(t *testing.T) {
```

Comment:
It may be better to break this up a little, to allow parallel test execution (via `t.Parallel()`),.

Reply:
Ok, I can try that.